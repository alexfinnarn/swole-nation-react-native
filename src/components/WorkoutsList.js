import React from 'react';
import {StyleSheet, FlatList, View} from "react-native";
import WorkoutListItem from "./WorkoutListItem";
import {styles} from './Styles';

export default function WorkoutsList({ navigation, workouts }) {
  // Add key that is needed for rendering the <FlatList /> component.
  workouts = workouts.map((workout) => {
    workout.key = workout.id.toString();
    return workout;
  });

  return (
    <View style={styles.container}>
      <FlatList
        data={workouts}
        renderItem={(item) => <WorkoutListItem workout={item} navigation={navigation}/>}
      />
    </View>
  )
}

