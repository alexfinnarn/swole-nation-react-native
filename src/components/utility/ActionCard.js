import React from 'react';
import {Text, View} from 'react-native';
import {home} from '../Styles';
import ActionButton from './ActionButton';

export default function ActionCard({children, actionComponent}) {
  return (
    <View style={home.sectionContainer}>
      <View style={home.sectionLeft}>{children}</View>
      {actionComponent}
    </View>
  );
}
