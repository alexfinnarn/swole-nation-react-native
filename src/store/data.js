import addFive from '../transformers/addFive'

const data = {
  workouts: {
    DQkECwYLCQQ: {
      name: "Stronglifts A",
      key: "DQkECwYLCQQ",
      description: "Squats, Bench Press, Barbell Row",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Bench Press Warmup",
        "Bench Press",
        "Barbell Row Warmup",
        "Barbell Row"
      ]
    },
    DAwJBQEMAA0: {
      name: "Stronglifts B",
      key: "DAwJBQEMAA0",
      description: "Squats, Overhead Press, Deadlifts",
      exercises: [
        "Squats Warmup",
        "Squats",
        "Overhead Press Warmup",
        "Overhead Press",
        "Deadlifts Warmup",
        "Deadlifts"
      ]
    }
  },
  exercises: {
    BwEBDwoFBgQ: {
      name: "Squats Warmup",
      key: "BwEBDwoFBgQ",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    },
    Bw0ECwAKBws: {
      name: "Squats",
      key: "Bw0ECwAKBws",
      image: require("../../assets/squats.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        },
        {
          "reps": 5,
          "weight": 230.0
        }
      ]
    },
    BwwBDAIJDgw: {
      name: "Bench Press Warmup",
      key: "BwwBDAIJDgw",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 3,
          "weight": 135.0
        }
      ]
    },
    DgcBCA8BCQo: {
      name: "Bench Press",
      key: "DgcBCA8BCQo",
      image: require("../../assets/bench-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        },
        {
          "reps": 5,
          "weight": 165.0
        }
      ]
    },
    DwcACw4PCgI: {
      name: "Shoulder Press Warmup",
      key: "DwcACw4PCgI",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 3,
          "weight": 75.0
        }
      ]
    },
    CAMBAwsFBwQ: {
      name: "Shoulder Press",
      key: "CAMBAwsFBwQ",
      image: require("../../assets/shoulder-press.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        },
        {
          "reps": 5,
          "weight": 110.0
        }
      ]
    },
    BwYGAw0NDwo: {
      name: "Barbell Row Warmup",
      key: "BwYGAw0NDwo",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 95.0
        }
      ]
    },
    BwQKAwYKAgA: {
      name: "Barbell Row",
      key: "BwQKAwYKAgA",
      image: require("../../assets/barbell-row.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 115.0
        },
        {
          "reps": 5,
          "weight": 115.0
        }
      ]
    },
    AQcNAQkBCwo: {
      name: "Deadlifts Warmup",
      key: "AQcNAQkBCwo",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 5,
          "weight": 185.0
        },
        {
          "reps": 5,
          "weight": 205.0
        }
      ]
    },
    DAYCDAQKAw4: {
      name: "Deadlifts",
      key: "DAYCDAQKAw4",
      image: require("../../assets/deadlift.gif"),
      instructions: "Do with a barbell",
      sets: [
        {
          "reps": 5,
          "weight": 225.0
        }
      ]
    }
  },
  sessions: {
    BwgFDQ8CCg8: {
      name: "new ...date",
      key: "BwgFDQ8CCg8",
      workoutKeys: [],
      exercises: []
    }
  },
  transformers: {
    none: {
      key: 'none',
      label: "N/A",
      callback: null
    },
    addFive: {
      key: 'addFive',
      label: "Exercises (+5 lbs)",
      callback: addFive
    }
  },
  activeWorkoutKey: "",
  activeExerciseKey: "",
  activeTransformerKey: "none",
};

export default data;
