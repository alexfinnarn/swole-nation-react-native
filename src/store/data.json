{
  "workouts": [
    {
      "name": "Stronglifts A",
      "key": "DQkECwYLCQQ",
      "description": "Squats, Bench Press, Barbell Row",
      "exercises": [
        "Squats Warmup"
      ]
    }
  ],
  "exercises": [
    {
      "name": "Squats Warmup",
      "key": "BwEBDwoFBgQ",
      "image": "../../assets/squats.gif",
      "instructions": "Do with a barbell",
      "sets": [
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 45.0
        },
        {
          "reps": 5,
          "weight": 95.0
        },
        {
          "reps": 5,
          "weight": 135.0
        },
        {
          "reps": 3,
          "weight": 185.0
        },
        {
          "reps": 2,
          "weight": 205.0
        }
      ]
    }
  ],
  "sessions": [
    {
      "name": "new ...date",
      "key": "BwgFDQ8CCg8",
      "workoutKeys": [],
      "exercises": []
    }
  ],
  "tempExerciseList": [],
  "theThing": "",
  "activeWorkoutIndex": 0,
  "activeSessionIndex": 0,
  "activeExerciseIndex": 0
}
