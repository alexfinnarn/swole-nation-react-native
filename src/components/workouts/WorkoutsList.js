import React from 'react';
import {FlatList, Text, View} from "react-native";
import {home, styles} from '../Styles';
import ActionButton from "../utility/ActionButton";
import ActionCard from "../utility/ActionCard";

export default function WorkoutsList({workouts, navigation, handle, thing}) {
  return (
    <View style={styles.container} testID="workouts-list-root">
      <View style={{flex: 9}}>
        <FlatList
          extraData={thing}
          data={workouts}
          renderItem={(item) => <WorkoutsListItem workout={item}/>}
        />
      </View>
      {/*<ActionButton label="Add new workout" text="Add Workout" action={() => {*/}
      {/*  handle.createWorkout();*/}
      {/*  navigation.navigate('Workout', {action: 'save'});*/}
      {/*}}/>*/}
    </View>
  );

  function WorkoutsListItem({workout}) {
    {/*  <ActionButton label={`Delete ${workout.item.name} workout`} text="X" action={() => {*/}
    {/*    handle.deleteWorkout(workout.item.key);*/}
    {/*  }}/>*/}
    {/*</>*/}

    return (
      <ActionCard actionComponent={
        <ActionButton label={`Edit ${workout.item.name} workout`} text="Edit" styles={{marginRight: 2}} action={() => {
          handle.setActiveWorkoutKey(workout.item.key);
          navigation.navigate('Workout', {
            workoutKey: workout.item.key,
            title: workout.item.name,
            action: 'edit'
          });
        }}/>
      }>
        <Text style={home.sectionHeaderText}>{workout.item.name}</Text>
        <Text style={{flex: 4, marginTop: 15}}>{workout.item.description}</Text>
      </ActionCard>
    );
  }
}

