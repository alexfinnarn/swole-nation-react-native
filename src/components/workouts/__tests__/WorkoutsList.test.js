import React from 'react';
import WorkoutsList from '../WorkoutsList';
import { render, fireEvent } from 'react-native-testing-library';
import data from '../../../store/data';
import ActionCard from "../../utility/ActionCard";
import shortId from "shortid";

let renderer, instance ={};
let workouts = Object.keys(data.workouts).map(key => data.workouts[key]);
const handle = {
  createWorkout: jest.fn(),
  deleteWorkout: jest.fn(),
  setActiveWorkoutKey: jest.fn(),
};
const navigation = {
  navigate: jest.fn()
};

beforeEach(() => {
  renderer = render(<WorkoutsList workouts={workouts} navigation={navigation} handle={handle} thing={shortId.generate()}/>);
  instance = renderer.getByTestId('workouts-list-root');
}, 0);

describe('<WorkoutsList />', () => {
  // @todo Foiled by the extraData param that is always unique. I shouldn't need that data to re-render so I should remove it.
  // it('Renders correctly', () => {
  //   expect(renderer.toJSON()).toMatchSnapshot();
  // });

  it('Should render two workouts in the list initially and three when workout added', () => {
    let actionCards = instance.findAllByType(ActionCard);
    expect(actionCards.length).toBe(2);
    expect(renderer.getAllByText('Edit').length).toBe(2);

    workouts.push({key: 'DWYVDAQWAw4', name: 'Workout C', description: 'blah', exercises: []});
    renderer.update(<WorkoutsList workouts={workouts} navigation={navigation} handle={handle} thing={shortId.generate()}/>);
    instance = renderer.getByTestId('workouts-list-root');

    actionCards = instance.findAllByType(ActionCard);
    expect(actionCards.length).toBe(3);
    expect(renderer.getAllByText('Edit').length).toBe(3);
  });

  it('Should navigate to the Workout screen when edit button pressed', () => {
    // Expect navigate to Stronglifts A workout.
    fireEvent(renderer.getAllByText('Edit')[0], 'press');
    expect(navigation.navigate).toHaveBeenCalledWith('Workout', {action: 'edit', workoutKey: 'DQkECwYLCQQ'});
    expect(handle.setActiveWorkoutKey).toHaveBeenCalledTimes(1);
    expect(handle.setActiveWorkoutKey).toHaveBeenCalledWith('DQkECwYLCQQ');


    // Expect navigate to Stronglifts B workout.
    fireEvent(renderer.getAllByText('Edit')[1], 'press');
    expect(navigation.navigate).toHaveBeenCalledWith('Workout', {action: 'edit', workoutKey: 'DAwJBQEMAA0'});
    expect(handle.setActiveWorkoutKey).toHaveBeenCalledTimes(2);
    expect(handle.setActiveWorkoutKey).toHaveBeenCalledWith('DAwJBQEMAA0');
  });

  // it('Should delete a workout and remove it from the list', () => {
  //   expect(renderer.queryByText('Stronglifts A')).not.toBeNull();
  //   fireEvent(renderer.getAllByText('X')[0], 'press');
  //   expect(handle.deleteWorkout).toHaveBeenCalledTimes(1);
  //   expect(handle.deleteWorkout).toHaveBeenCalledWith('DQkECwYLCQQ');
  //
  //   workouts.shift();
  //   renderer.update(<WorkoutsList workouts={workouts} navigation={navigation} handle={handle} thing={shortId.generate()}/>);
  //   expect(renderer.queryByText('Stronglifts A')).toBeNull();
  // });
});
