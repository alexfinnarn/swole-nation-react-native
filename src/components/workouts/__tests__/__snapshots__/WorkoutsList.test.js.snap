// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WorkoutsList /> Renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "flexDirection": "column",
      "paddingTop": 5,
    }
  }
  testID="workouts-list-root"
>
  <View
    style={
      Object {
        "flex": 9,
      }
    }
  >
    <RCTScrollView
      data={
        Array [
          Object {
            "description": "Squats, Bench Press, Barbell Row",
            "exercises": Array [
              "Squats Warmup",
              "Squats",
              "Bench Press Warmup",
              "Bench Press",
              "Barbell Row Warmup",
              "Barbell Row",
            ],
            "key": "DQkECwYLCQQ",
            "name": "Stronglifts A",
          },
          Object {
            "description": "Squats, Overhead Press, Deadlifts",
            "exercises": Array [
              "Squats Warmup",
              "Squats",
              "Overhead Press Warmup",
              "Overhead Press",
              "Deadlifts Warmup",
              "Deadlifts",
            ],
            "key": "DAwJBQEMAA0",
            "name": "Stronglifts B",
          },
        ]
      }
      disableVirtualization={false}
      extraData="qXSkwuSi"
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Stronglifts A
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Squats, Bench Press, Barbell Row
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  Object {
                    "marginRight": 2,
                  },
                ]
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              Object {
                "backgroundColor": "#ffffff",
                "elevation": 5,
                "flex": 1,
                "flexDirection": "row",
                "marginBottom": 4,
                "marginTop": 4,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 2,
                  "width": 0,
                },
                "shadowOpacity": 0.25,
                "shadowRadius": 3.84,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 4,
                  "flexDirection": "column",
                  "padding": 10,
                }
              }
            >
              <Text
                style={
                  Object {
                    "flex": 1,
                    "fontSize": 18,
                  }
                }
              >
                Stronglifts B
              </Text>
              <Text
                style={
                  Object {
                    "flex": 4,
                    "marginTop": 15,
                  }
                }
              >
                Squats, Overhead Press, Deadlifts
              </Text>
            </View>
            <View
              style={
                Array [
                  Object {
                    "flex": 1,
                    "flexDirection": "column",
                  },
                  Object {
                    "marginRight": 2,
                  },
                ]
              }
            >
              <View
                accessible={true}
                isTVSelectable={true}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "alignSelf": "stretch",
                    "backgroundColor": "#21897E",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "alignSelf": "center",
                        "color": "#ffffff",
                        "fontSize": 20,
                      }
                    }
                  >
                    Edit
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
